include/utransport/UAttributes.h

```diff
@@ -0,0 +1,299 @@
+                                    attributes_->id_ = id;
+                                    attributes_->type_ = type;
+                                    attributes_->priority_ = priority;
+                                }
+                            }
+
+                            UAttributesBuilder& withTtl(int ttl) {
+
+                                attributes_->ttl_ = ttl;
+                                return *this;
+                            }
+
+                            UAttributesBuilder& withToken(const std::string& token) {
+
+                                attributes_->token_ = token;
+                                return *this;
+                            }
+
+                            UAttributesBuilder& withHint(const USerializationHint& hint) {
+
+                                attributes_->hint_ = hint;
+                                return *this;
+                            }
+
+                            UAttributesBuilder& withSink(const UUri& sink) {
+
+                                std::optional<UUri> optionalUri = sink; 
+                                attributes_->sink_ = &optionalUri;
+                                return *this;
+                            }
+
+                            UAttributesBuilder& withPermissionLevel(int plevel) {
+
+                                attributes_->plevel_ = plevel;
+                                return *this;
+                            }
+
+                            UAttributesBuilder& withCommStatus(int commstatus) {
+
+                                attributes_->commstatus_ = commstatus;
+                                return *this;
+                            }
+
+                            UAttributesBuilder& withReqId(const UUID& reqid) {
+
+                                attributes_->reqid_ = reqid;
+                                return *this;
+                            }
+
+                            UAttributes build() {
+
+                                return *attributes_;
+                            }
+
+                        private:
+                            std::shared_ptr<UAttributes> attributes_;
+                    };
+                    
+                private:
+
+                    UUID id_;                                   // Unique identifier for the message
+                    UMessageType type_;                         // Message type
+                    UPriority priority_;                        // Message priority
+                    // Optional Attributes 
+                    std::optional<int32_t> ttl_;                // Time to live in milliseconds
+                    std::optional<std::string> token_;          // Authorization token used for TAP
+                    std::optional<USerializationHint> hint_;    // Hint regarding the bytes contained within the UPayload
+                    std::optional<UUri> *sink_;                 // Explicit destination URI
```

Comment:
can we use smart pointer ?

Reply:
can we use something like:
``c++
std::optional<std::unique_ptr<UUri>> sink_;
``